benchopt/utils/misc.py

```diff
@@ -84,3 +85,23 @@ def get_benchopt_requirement(pytest=False):
             )
 
     return req, False
+
+
+def OSSpecificNamedTempFile(dir=None, mode='w+b', prefix=None, suffix=None):
```

Comment:
can you add the explanation you mad ein the first PR message here, so that future users understand why this is needed?